data/data/azure/vnet/main.tf

```diff
@@ -91,27 +91,75 @@ resource "azurerm_storage_blob" "rhcos_image" {
   metadata               = tomap({ source_uri = var.azure_image_url })
 }
 
-resource "azurerm_image" "cluster" {
+# Creates Shared Image Gallery
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/shared_image_gallery
+resource "azurerm_shared_image_gallery" "sig" {
+  name                = "gallery_${replace(var.cluster_id, "-", "_")}"
+  resource_group_name = data.azurerm_resource_group.main.name
+  location            = var.azure_region
+}
+
+# Creates image definition
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/shared_image
+resource "azurerm_shared_image" "cluster" {
   name                = var.cluster_id
+  gallery_name        = azurerm_shared_image_gallery.sig.name
   resource_group_name = data.azurerm_resource_group.main.name
   location            = var.azure_region
+  os_type             = "Linux"
 
-  os_disk {
-    os_type  = "Linux"
-    os_state = "Generalized"
-    blob_uri = azurerm_storage_blob.rhcos_image.url
+  identifier {
+    publisher = "RedHat"
+    offer     = "rhcos"
+    sku       = "basic"
   }
 }
 
-resource "azurerm_image" "clustergen2" {
+resource "azurerm_shared_image" "clustergen2" {
   name                = "${var.cluster_id}-gen2"
+  gallery_name        = azurerm_shared_image_gallery.sig.name
   resource_group_name = data.azurerm_resource_group.main.name
   location            = var.azure_region
+  os_type             = "Linux"
   hyper_v_generation  = "V2"
+  architecture        = var.azure_vm_architecture
 
-  os_disk {
-    os_type  = "Linux"
-    os_state = "Generalized"
-    blob_uri = azurerm_storage_blob.rhcos_image.url
+  identifier {
+    publisher = "RedHat-gen2"
+    offer     = "rhcos-gen2"
+    sku       = "gen2"
   }
 }
+
+resource "azurerm_shared_image_version" "cluster_image_version" {
+  name                = "0.0.1"
```

Comment:
I recognize that setting an arbitrary version is simpler, but we do actually know the rhcos version and it would be preferable to have the actual version here. 

I can think of two ways to get the version.

1. The RHCOS stream contains the `release`:
https://github.com/openshift/installer/blob/b8714c6d45d2700dcb44ddd86d2afb309e89fa35/pkg/asset/rhcos/image.go#L139
Similarly to the way we plumb the URL through the code base, we would also plumb the release.
2. The URL itself contains the version:
https://github.com/openshift/installer/blob/b8714c6d45d2700dcb44ddd86d2afb309e89fa35/data/data/coreos/rhcos.json#L768
It is a bit gnarly, but the version could be parsed from the URL

Reply:
ah..ok..i'll look into this. i am thinking I'll do a follow up PR for this if i find out it involves a bit of change..

Reply:
well..i did try and make the changes but looks like the image gallery version names are pretty strict:
```
ERROR Error: Expected name to be in the format `1.2.3`, `latest`, or `recent` but got "412.86.202208101039". 
ERROR                                              
ERROR   with azurerm_shared_image_version.cluster_image_version, 
ERROR   on main.tf line 139, in resource "azurerm_shared_image_version" "cluster_image_version": 
ERROR  139:   name                = var.azure_image_release 

```
i removed the `-0` thinking that was the issue but still got this error. so looks like it only allows single digit versions

Reply:
we could use `latest` like the error says rather than an arbitrary string

Reply:
@patrickdillon thoughts ?

Reply:
Good idea: let's use latest.

Reply:
tried to use "latest", but then it wasn't happy with that either:
```
ERROR Error: creating Shared Image Version: (Version Name "latest" / Image Name "psundara-ocp-c99wz-gen2" / Gallery Name "gallery_psundara_ocp_c99wz" / Resource Group "psundara-ocp-c99wz-rg"): compute.GalleryImageVersionsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="The image version name is invalid. The image version name should follow Major(int).Minor(int).Patch(int) format, where int is between 0 and 2,147,483,647 (inclusive). For e.g: 1.0.0, 2018.12.1 etc." Target="galleryImageVersionName"
``` 
but then the error message does say that the numbers must be between 0-2,147483647 so i just truncated the date in the rhcos string and was able to use the release image string

Reply:
used the rhcos release version by truncating the date